C-Plus-Plus/other/addTwoNumbers.cpp

```diff
@@ -0,0 +1,160 @@
+        int sum = carry + x + y; // calculating sum of the respective nodes
+        carry = sum / 10; // calculating the carry
+        Insertion(answer_list,(sum % 10)); // storing the sum obtained in the answerList;
+        if (pointer1 != NULL)
+            pointer1 = pointer1->next;
+        if (pointer2 != NULL)
+            pointer2 = pointer2->next;
+    }
+    if (carry > 0)
+        Insertion(answer_list,carry); 
+    return answer_list;
+}
+node* ReverseTheList(node* &head){
+    node* prev_ptr = NULL;
+    node* current_ptr = head;
+    node* next_ptr;
+    while(current_ptr != NULL)
+    {
+        next_ptr = current_ptr -> next;
+        current_ptr -> next = prev_ptr;
+        prev_ptr = current_ptr;
+        current_ptr = next_ptr;
+    }
+    return prev_ptr;
+}
+// main function
+int main()
+{
+    int testCases;
+    cout << "Enter the number of testcases: ";
+    cin >> testCases;
+
+    while (testCases--)
+    {
+        node* input1 = NULL; // intialization of the input list1 
+        node* input2 = NULL; // intialintialization of the input list2
+        int number1;
+        cout << "Enter the size of the linked list 1: "<<endl;
+        cin >> number1;
+        int number2;
+        cout << "Enter the size of the linked list 2: "<<endl; 
+        cin>>number2;
+        cout<<"Enter the elements of linked list 1: "<<endl;
+        while(number1--)
+        {
+            int temp;
+            cin>>temp;
+            Insertion(input1, temp);
+        }
+        cout << "Enter the elements of linked list 2: " << endl;
+        while(number2--){
+            int temp;
+            cin>>temp;
+            Insertion(input2,temp);
+        }
+        node* ans = AddTwoNumbers(input1, input2);
+        ans = ReverseTheList(ans);
+        cout<<"The resulting sum of the two lists is: "<<endl;
+        while(ans != NULL)
+        {
+            cout<<ans->value<<"";
+            ans = ans->next;
+        }
+        cout<<endl;
+    }
+
+    return 0;
+}
+
+/*
+
+FOR EXAMPLE:--
+
+Example 1:
+Input:-
+Enter the number of testcases: 3
+Enter the size of the linked list 1: 
+7
+Enter the size of the linked list 2: 
+4
+Enter the elements of linked list 1: 
+9 9 9 9 9 9 9
+Enter the elements of linked list 2: 
+9 9 9 9
+The resulting sum of the two lists is: 
+10009998
+Enter the size of the linked list 1: 
+3
+Enter the size of the linked list 2: 
+3
+Enter the elements of linked list 1: 
+2 4 3
+Enter the elements of linked list 2: 
+5 6 4
+The resulting sum of the two lists is: 
+807
+Enter the size of the linked list 1: 
+1
+Enter the size of the linked list 2: 
+1
+Enter the elements of linked list 1: 
+0
+Enter the elements of linked list 2: 
+0
+The resulting sum of the two lists is: 
+0
+
+
+SPACE COMPLEXITY = O(max(m,n))(where m and n are the sizes of the list1 and list2 respectively)
+TIME COMPLEXITY = O(max(m,n))
+
+*/
```

Comment:
You need to add a line at EOF. Rest everything is good.

Reply:
Can u tell what is an EOF